validator_client/beacon_node_fallback/src/lib.rs

```diff
@@ -646,6 +688,39 @@ impl<T: SlotClock> BeaconNodeFallback<T> {
         Err(Errors(errors))
     }
 
+    /// Try `func` on a specific beacon node by index first, then fall back to the normal order.
+    /// Returns immediately if the preferred node succeeds, otherwise falls back to first_success.
+    /// This is an insurance against potential race conditions that may arise.
+    pub async fn first_success_from_index<F, O, Err, R>(
+        &self,
+        preferred_index: Option<usize>,
+        func: F,
+    ) -> Result<O, Errors<Err>>
+    where
+        F: Fn(BeaconNodeHttpClient) -> R + Clone,
+        R: Future<Output = Result<O, Err>>,
+        Err: Debug,
+    {
+        // Try the preferred beacon node first if it exists
+        if let Some(preferred_idx) = preferred_index
+            && let candidates = self.candidates.read().await
+            && let Some(preferred_candidate) = candidates.iter().find(|c| c.index == preferred_idx)
+        {
+            let preferred_node = preferred_candidate.beacon_node.clone();
+            drop(candidates);
+
+            match Self::run_on_candidate(preferred_node, &func).await {
+                Ok(val) => return Ok(val),
+                Err(_) => {
+                    return self.first_success(func).await;
+                }
+            }
+        }
+
+        // Fall back to normal first_success behavior
```

Comment:
We should log a debug message here if the preferred candidate is not found, as this should be very unlikely once we use IDs rather than indices

Reply:
@macladson Is it intentional that the current `index` is immutable? I noticed that it isn't updated as part of `sort_nodes_by_health`?

https://github.com/hopinheimer/lighthouse/blob/409d937ed2e9eea224da9688299b834c4f674544/validator_client/beacon_node_fallback/src/lib.rs#L804-L827

So it doesn't correspond to the `index` of the beacon node within the vec. Maybe we can rename that `index` to `id` to make this clearer.